src/utils/series-utils.js

```diff
@@ -71,41 +74,47 @@ function collectSeriesTypesInfo(children) {
  * Each next value of attr is equal to the previous value plus the difference
  * between attr0 and attr.
  * @param {Array} children Array of children.
- * @param {string} attr Attribute.
+ * @param {string} attr Attribute to stack by.
  * @returns {Array} New array of children for the series.
  */
 export function getStackedData(children, attr) {
-  const childData = [];
-  let prevIndex = -1;
-  children.forEach((child, childIndex) => {
+  return children.reduce((accumulator, series, seriesIndex) => {
     // Skip the children that are not series (e.g. don't have any data).
-    if (!child) {
-      childData.push(null);
-      return;
+    if (!series) {
+      accumulator.result.push(null);
+      return accumulator;
     }
-    const {data} = child.props;
+
+    const {data, cluster = 'default'} = series.props;
```

Comment:
Yeah - assuming that you're using `stackBy` it will always go through the clustering code path. I guess this is a bit suboptimal, but there's no way of telling if sibling series have a `cluster` key, unless did an extra loop at the start to determine this.